### ISSUE TITLE
Positive signed integers not handled properly

### ISSUE THREAD

**Issue by aditsachde** [ipld/libipld#110]
Hey, this crate has been super useful, but I came across a bit of an issue while using it.

In the CBOR [spec](https://www.rfc-editor.org/rfc/rfc8949#section-3.1), an unsigned integer is defined as an integer in the range `0..2^64-1` inclusive. A negative integer is defined as an integer in the range `-2^64..-1` inclusive. Therefore, when encoding a positive isize, it should have the CBOR major type 0 (unsigned).

The following code outputs `a1, 67, 69, 6e, 74, 65, 67, 65, 72, 3a, ff, ff, ff, f5` which when decoded with [another cbor decoder](http://cbor.me/) outputs `{"integer": -4294967286}` because the 10 gets converted into `negative(4294967285)` instead of `unsigned(10)`.

```rust
struct test {
    integer: i32
}

fn main() {
    let test = test {
        integer: 10
    };
    let result = DagCborCodec.encode(&test).unwrap();
    println!("{:x?}", result);
}
```

I'm intending to open a PR to fix this, but it might be a bit before I get around to it.

**Comment by dvc94ch** [comment]
Have you checked with the serde cbor implementation? But this does seem bad.

**Comment by dvc94ch** [comment]
Can you open a PR to fix it? It's not quite as bad as a roundtrip does work.

**Comment by molekilla** [comment]
related https://github.com/ipfs-rust/libipld/issues/113

**Comment by Stebalien** [comment]
Fixed in #125 but not yet merged to master.

**Comment by vmx** [comment]
This is fixed with the merge of #129.
